stdlib/Dates/src/arithmetic.jl

```diff
@@ -6,8 +6,8 @@
 
 # TimeType arithmetic
 (+)(x::TimeType) = x
-(-)(x::T, y::T) where {T<:TimeType} = x.instant - y.instant
-(-)(x::TimeType, y::TimeType) = -(promote(x, y)...)
+(-)(x::T, y::T) where {T<:Union{AbstractDateTime,TimeType}} = x.instant - y.instant
```

Comment:
I don't think having an `instant` field is part of the contract of `AbstractDateTime`. Moreover,  `AbstractDateTime <:  TimeType` already, so the Union collapses to that:

```
julia> Union{Dates.AbstractDateTime,Dates.TimeType}
TimeType
```